haystack_experimental/dataclasses/tool.py

```diff
@@ -198,6 +205,59 @@ def get_weather(
 
         return Tool(name=name or function.__name__, description=tool_description, parameters=schema, function=function)
 
+    @classmethod
+    def from_component(cls, component: Component, name: str, description: str) -> "Tool":
+        """
+        Create a Tool instance from a Haystack component.
+
+        :param component: The Haystack component to be converted into a Tool.
+        :param name: Name for the tool.
```

Comment:
I don't want to make this PR more complex but have we considered using defaults for name and description? For example the component name as the default for `name` and parts of the component docstring extracted via docstring_parser for `description`? 
That would also enable the ToolInvoker to call from_component internally if the input to its `tool` parameter is a list of components instead of tools, which I think is needed for defining tools as components in a yaml.